tests/python/pants_test/backend/python/interpreter_selection_utils.py

```diff
@@ -107,3 +110,48 @@ def skip_unless_python27_and_python3_present(func):
 def skip_unless_python27_and_python36_present(func):
   """A test skip decorator that only runs a test method if python2.7 and python3.6 are present."""
   return skip_unless_all_pythons_present(PY_27, PY_36)(func)
+
+
+def skip_if_interpreter_is_any_python_version(*versions):
+  """A decorator that skips if the current interpreter version is any of the of the specified versions.
+
+  :param string *versions: Python version strings, such as "2.7", "3".
+  """
+  current_interpreter = ".".join(map(str, sys.version_info[0:2]))
+  # NB: if only the major Python version is specified, e.g. `2` or `3`, we must append `.*` to the string
```

Comment:
Would it be a useful validation to attempt to parse these as floating point, and declare them invalid otherwise? I could imagine someone trying to pass in something like `CPython>=3`. ...but maybe that would fail with a good error in `SpecifierSet`?

Reply:
`SpecifierSet` performs great validation for us already. Running `pex --python=python3 packaging==16.8`:

```python
>>> from packaging.specifiers import SpecifierSet
>>> SpecifierSet("CPython>3.6")
Traceback (most recent call last):
  File "/private/var/folders/sx/pdpbqz4x5cscn9hhfpbsbqvm0000gn/T/tmpvkf49eb1/.deps/packaging-16.8-py2.py3-none-any.whl/packaging/specifiers.py", line 601, in __init__
    parsed.add(Specifier(specifier))
  File "/private/var/folders/sx/pdpbqz4x5cscn9hhfpbsbqvm0000gn/T/tmpvkf49eb1/.deps/packaging-16.8-py2.py3-none-any.whl/packaging/specifiers.py", line 85, in __init__
    raise InvalidSpecifier("Invalid specifier: '{0}'".format(spec))
packaging.specifiers.InvalidSpecifier: Invalid specifier: 'CPython>3.6'
```